File: src/main/java/com/jozufozu/exnihiloomnia/common/items/ExNihiloMaterials.java
package com.jozufozu.exnihiloomnia.common.items;

import net.minecraft.item.Item.ToolMaterial;
import net.minecraft.item.ItemStack;
import net.minecraftforge.common.util.EnumHelper;
import net.minecraftforge.fml.common.registry.GameRegistry;

public class ExNihiloMaterials
{
    public static ToolMaterial SILK;
    public static ToolMaterial SKELETAL;
    
    @GameRegistry.ItemStackHolder("minecraft:string")
    public static ItemStack STRING;
    
    @GameRegistry.ItemStackHolder("minecraft:bone")
    public static ItemStack BONE;
    
    public static void preInit()
    {
        SILK = EnumHelper.addToolMaterial("SILK", 0, 128, 10.0f, 0, 18);
        SKELETAL = EnumHelper.addToolMaterial("BONE", 1, 200, 8.0f, 2, 25);
    }
    
    public static void init()
    {
        SILK.setRepairItem(STRING);
        SKELETAL.setRepairItem(BONE);
    }
}

File: src/main/java/com/jozufozu/exnihiloomnia/common/items/ExNihiloMaterials.kt
package com.jozufozu.exnihiloomnia.common.items

import net.minecraft.init.Items
import net.minecraft.item.Item.ToolMaterial
import net.minecraft.item.ItemStack
import net.minecraftforge.common.util.EnumHelper
import net.minecraftforge.fml.common.registry.GameRegistry

object ExNihiloMaterials {
    val SILK: ToolMaterial = EnumHelper.addToolMaterial("ENO_SILK", 0, 128, 10.0f, 0f, 18) ?: throw Exception("Error adding SILK to EnumToolMaterials")
    val SKELETAL: ToolMaterial = EnumHelper.addToolMaterial("ENO_BONE", 1, 200, 8.0f, 2f, 25) ?: throw Exception("Error adding BONE to EnumToolMaterials")

    private val STRING: ItemStack by lazy { ItemStack(Items.STRING) }
    private val BONE: ItemStack by lazy { ItemStack(Items.BONE) }

    fun preInit() {}

    fun init() {
        SILK.setRepairItem(STRING)
        SKELETAL.setRepairItem(BONE)
    }
}
